src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,330 @@
+  equations: a practical guide
+  [CERFACS Technical report](https://www.cerfacs.fr/~montagna/TR-CFD-13-77.pdf)
+The scaling used herein is Section 4.5 of the reference.
+"""
+struct CompressibleNavierStokesEquations2D{RealT <: Real, E <: AbstractCompressibleEulerEquations{2}} <: AbstractCompressibleNavierStokesEquations{2, 4}
+  # TODO:
+  # 1) For now save gamma and inv(gamma-1) again, but could potentially reuse them from the Euler equations
+  # 2) Add NGRADS as a type parameter here and in AbstractEquationsParabolic, add `ngradients(...)` accessor function
+  gamma::RealT               # ratio of specific heats
+  inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+  Re::RealT                  # Reynolds number
+  Pr::RealT                  # Prandtl number
+  Ma_inf::RealT              # free-stream Mach number
+  kappa::RealT               # thermal diffusivity for Fick's law
+
+  p_inf::RealT               # free-stream pressure
+  u_inf::RealT               # free-stream velocity
+  R::RealT                   # gas constant (depends on nondimensional scaling!)
+
+  equations_hyperbolic::E    # CompressibleEulerEquations2D
+end
+
+function CompressibleNavierStokesEquations2D(equations::CompressibleEulerEquations2D; Reynolds, Prandtl, Mach_freestream)
+  γ = equations.gamma
+  inv_gamma_minus_one = equations.inv_gamma_minus_one
+  Re, Pr, Ma = promote(Reynolds, Prandtl, Mach_freestream)
+
+  # Under the assumption of constant Prandtl number the thermal conductivity
+  # constant is kappa = gamma μ R / ((gamma-1) Pr).
+  # Important note! Due to nondimensional scaling R = 1 / gamma, this constant
+  # simplifies slightly. Also, the factor of μ is accounted for later.
+  κ = inv_gamma_minus_one / Pr
+
+  # From the nondimensionalization discussed above set the remaining free-stream
+  # quantities
+  p_inf = 1 / γ
+  u_inf = Mach_freestream
+  R     = 1 / γ
+  CompressibleNavierStokesEquations2D{typeof(γ),typeof(equations)}(γ, inv_gamma_minus_one,
+                                                                   Re, Pr, Ma, κ,
+                                                                   p_inf, u_inf, R,
+                                                                   equations)
+end
+
+
+# This is the flexibility a user should have to select the different gradient variable types
+# varnames(::typeof(cons2prim)   , ::CompressibleNavierStokesEquations2D) = ("v1", "v2", "T")
+# varnames(::typeof(cons2entropy), ::CompressibleNavierStokesEquations2D) = ("w2", "w3", "w4")
+
+varnames(variable_mapping, equations_parabolic::CompressibleNavierStokesEquations2D) =
+  varnames(variable_mapping, equations_parabolic.equations_hyperbolic)
+
+# we specialize this function to compute gradients of primitive variables instead of
+# conservative variables.
+gradient_variable_transformation(::CompressibleNavierStokesEquations2D, dg_parabolic) = cons2prim
```

Comment:
I left this here because I wasn't sure how we wanted to dispatch on different formulations of the Navier-Stokes equations. 

Should we have different parabolic equation types for standard formulation vs entropy variable formulation vs other formulations (e.g., we're working on a paper on a split form) of Navier-Stokes diffusion? I thought these different formulations might be encoded in `dg_parabolic`, but I'm not sure about this.

Reply:
>  but I'm not sure about this

For this very reason I would just leave it out for now until we have a prototype that actually requires this kind of dispatch.

Reply:
Done in 144fc593a31502b5de39cd3d4ecaf77a586331c2